src/components/schedule-visualizer/colors.ts

```diff
@@ -21,29 +8,81 @@ async function _hash(s: string): Promise<ArrayBuffer> {
 
 export default class ColorManager {
   async robotColor(name: string, model: string): Promise<string> {
-    const key = this._robotKey(name, model);
-    let color = this._robotColorCache[key];
+    let color = this._robotColorCache[name];
     if (!color) {
       const modelHash = new Uint16Array(await _hash(model));
-      const hue = modelHash[0] % 360;
       const nameHash = new Uint16Array(await _hash(name));
-      const satlum = nameHash[0] % 2500;
-      const saturation = 50 + (satlum % 50);
-      const luminance = 25 + satlum / 50;
-      color = `hsl(${hue}, ${saturation}%, ${luminance}%)`;
-      this._robotColorCache[key] = color;
+      color = ColorManager._getLightColor(modelHash[0], nameHash[0]);
+      this._robotColorCache[name] = color;
     }
     return color;
   }
 
-  robotColorFromCache(name: string, model: string): string | null {
-    const color = this._robotColorCache[this._robotKey(name, model)];
-    return color ? color : null;
+  async robotTrajectoryColor(name: string, model: string): Promise<string> {
+    let color = this._pathColorCache[name];
+    if (!color) {
+      const modelHash = new Uint16Array(await _hash(model));
+      const nameHash = new Uint16Array(await _hash(name));
+      color = ColorManager._gePathColor(modelHash[0], nameHash[0]);
+      this._pathColorCache[name] = color;
+    }
+    return color;
   }
 
-  private _robotColorCache: Record<string, string> = {};
+  // use vibrant package
+  async robotImageColor(path: string, name: string) {
+    let color = this._robotColorCache[name];
+    if (!color) {
+      const imgHolder = new Image(400, 400);
+      imgHolder.src = path;
 
-  private _robotKey(name: string, model: string) {
-    return `${name}__${model}`;
+      await new Promise((resolve, reject) => {
+        imgHolder.onload = () => resolve();
+        imgHolder.onerror = err => reject(err);
+      });
+      return Vibrant.from(imgHolder)
+        .getSwatches()
+        .then(palette => {
+          const rgb = palette.Vibrant?.getRgb();
+          if (rgb) {
+            const colorHolder = `rgb(${rgb[0]}, ${rgb[1]}, ${rgb[2]})`;
+            this._robotColorCache[name] = colorHolder;
+            return colorHolder;
+          }
+        });
+    }
   }
+
+  robotColorFromCache(name: string): string | null {
```

Comment:
we might want to make this fleet + name to prevent the super rare case of a name clash.